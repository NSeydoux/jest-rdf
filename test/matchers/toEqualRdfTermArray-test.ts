import { DataFactory } from 'rdf-data-factory';
import "../../index";

const DF = new DataFactory();

describe('#toEqualRdfTermArray', () => {
  it('should succeed for equal term arrays', () => {
    return expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).toEqualRdfTermArray([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]);
  });

  it('should not fail for equal quad arrays', () => {
    return expect(() => expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).not.toEqualRdfTermArray([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not succeed for quad arrays with different length', () => {
    return expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).not.toEqualRdfTermArray([
      DF.namedNode('s1'),
    ]);
  });

  it('should fail for quad arrays with different length', () => {
    return expect(() => expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).toEqualRdfTermArray([
      DF.namedNode('s1'),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not succeed for quad arrays with equal length but different contents', () => {
    return expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).not.toEqualRdfTermArray([
      DF.namedNode('s2'),
      DF.namedNode('p2'),
      DF.namedNode('o2'),
      DF.namedNode('g2'),
    ]);
  });

  it('should fail for quad arrays with equal length but different contents', () => {
    return expect(() => expect([
      DF.namedNode('s1'),
      DF.namedNode('p1'),
      DF.namedNode('o1'),
      DF.namedNode('g1'),
    ]).toEqualRdfTermArray([
      DF.namedNode('s2'),
      DF.namedNode('p2'),
      DF.namedNode('o2'),
      DF.namedNode('g2'),
    ])).toThrowErrorMatchingSnapshot();
  });
});
