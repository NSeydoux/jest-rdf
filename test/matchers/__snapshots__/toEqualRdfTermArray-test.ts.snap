// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toEqualRdfTermArray should fail for quad arrays with different length 1`] = `"expected [ \\"s1\\", \\"p1\\", \\"o1\\", \\"g1\\" ] to equal [ \\"s1\\" ]"`;

exports[`#toEqualRdfTermArray should fail for quad arrays with equal length but different contents 1`] = `"expected s1 and s2 to be equal"`;

exports[`#toEqualRdfTermArray should fail for quad arrays with nested quads with equal length but different contents 1`] = `"expected <<s2 p o g>> and <<s2 p o g->> to be equal"`;

exports[`#toEqualRdfTermArray should not fail for equal quad arrays 1`] = `"expected [ \\"s1\\", \\"p1\\", \\"o1\\", \\"g1\\" ] not to equal [ \\"s1\\", \\"p1\\", \\"o1\\", \\"g1\\" ]"`;
